hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestAHSv2ClientImpl.java

```diff
@@ -107,6 +113,91 @@ public void testGetAppAttemptReport() throws IOException, YarnException {
+
+    // Multiple container test
+    final int multipleContainerSize = 4;
+    final int appMultipleId = 3;
+    final int appAttemptMultipleId = 3;
+    final ApplicationId appMultiple = ApplicationId.newInstance(0, appMultipleId);
+    final ApplicationAttemptId appAttemptIdMultiple =
+            ApplicationAttemptId.newInstance(appMultiple, appAttemptMultipleId);
+
+    List<TimelineEntity> containerEntities = new ArrayList<>();
+    for (int newEntityIdx = 0; newEntityIdx < multipleContainerSize; ++newEntityIdx) {
+      containerEntities.add(createContainerEntity(ContainerId.newContainerId(
+              appAttemptIdMultiple, containerId2.getContainerId() +1 + newEntityIdx)));
+    }
+
+    when(spyTimelineReaderClient.getContainerEntities(
+            appId1, "ALL",
+            ImmutableMap.<String, String>builder()
+                    .put("appattemptId", appAttemptId1.toString())
+                    .build(),
+            0, null))
+            .thenReturn(Arrays.asList(createContainerEntity(containerId1)));
+    when(spyTimelineReaderClient.getContainerEntities(
+            appId2, "ALL",
+            ImmutableMap.<String, String>builder()
+                    .put("appattemptId", appAttemptId2.toString())
+                    .build(),
+            0, null))
+            .thenReturn(Arrays.asList(createContainerEntity(containerId2)));
+
+    when(spyTimelineReaderClient.getContainerEntities(
+            appMultiple, "ALL",
+            ImmutableMap.<String, String>builder()
+                    .put("appattemptId", appAttemptIdMultiple.toString())
+                    .build(),
+            0, null))
+            .thenReturn(containerEntities);
+
+    when(spyTimelineReaderClient.getApplicationEntity(appId1, "ALL", null))
+            .thenReturn(createApplicationTimelineEntity(appId1, true,
+                    false));
+    when(spyTimelineReaderClient.getApplicationEntity(appId2, "ALL", null))
+            .thenReturn(createApplicationTimelineEntity(appId2, true,
+                    false));
+    when(spyTimelineReaderClient.getApplicationEntity(appMultiple, "ALL", null))
+            .thenReturn(createApplicationTimelineEntity(appMultiple, true,
+                    false));
+
+    List<ContainerReport> containerList1 = client.getContainers(appAttemptId1);
+    List<ContainerReport> containerList2 = client.getContainers(appAttemptId2);
+
+    List<ContainerReport> containerListMultiple = client.getContainers(appAttemptIdMultiple);
+
+    assertThat(containerList1.size()).isEqualTo(1);
+    assertThat(containerList2.size()).isEqualTo(1);
+    assertThat(containerListMultiple.size()).isEqualTo(multipleContainerSize);
+
+    assertThat(containerList1.get(0).getContainerId().getApplicationAttemptId()
+            .getApplicationId().getId()).isEqualTo(1);
+    assertThat(containerList2.get(0).getContainerId().getApplicationAttemptId()
+            .getApplicationId().getId()).isEqualTo(2);
```

Comment:
Can you please use a variable for the applicationAttemptId in the isEqualTo part? It's easier to read that way.